.github/scripts/rebase-finish.sh

```diff
@@ -4,13 +4,20 @@ source `dirname ${BASH_SOURCE[0]}`/config.sh
 
 REBASE_BRANCH=$1
 ORIGIN_BRANCH=$2
+BACKUP_BRANCH=$3
 
 git fetch --all --prune
 
+# Check if $ORIGIN_BRANCH didn't change.
+if [ "$(git rev-parse origin/$BACKUP_BRANCH)" != "$(git rev-parse origin/$ORIGIN_BRANCH)" ]; then
+    echo "$ORIGIN_BRANCH has changed!"
+    exit 1
```

Comment:
As discussed, the daily rebase threshold could be introduced now or later (something like 3 days) to exit with 0, even if it has not been rebased by the end

Reply:
Why? In which situtation one would like to hide the information that the daily rebases are not working? How would you detect those 3 days?

Reply:
> It might be expected behavior if daily rebased has not been finished in the last 2 days. 

It does not explain why it would be expected? Again, why would anyone would like to ignore the fact that daily rebases have stopped doing its thing?

Reply:
It might be by design expected to cancel the workflow if other changes have been merged. And if it happens on 3rd attempt, notify as an error.

Reply:
I am not sure if I understand the same thing: What would make sense is that if nobody touched the \`rebase-upstream\` branch for three days, the workflow would not be triggered at all, or it would be cancelled with error during some early job, e.g., in \`rebase-start.sh\`. If this is what you mean, the logic handling that is unrelated to the logic here and it would again require some sort of persistence of the information when the \`rebase-upstream\` branch was created.

Reply:
Yes, the two "clenup" commands (https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/pull/247/files#diff-1114e6f1c6dd47ee94177da1b364a2b0a344b05591161348f63c65efc93659bdR39) can be added here, if manual resolution of the condition check failure is not expected.

Reply:
Added cleanup of `$REBASE_BRANCH` and `$BACKUP_BRANCH` when difference betwen `$BACKUP_BRANCH` and `$ORIGIN_BRANCH` is detected.